### PR TITLE
S3 parquet sink cloudformation

### PR DESCRIPTION
**Pull request by akshayar**
*Issue #, if available:*

*Description of changes:*
S3ParquetSink example modified with CloudFormation template and additional instructions for local run.
Removed the need to hardcode access and secret key.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by akshayar** [comment]
@nicusX  @jeremyber-aws @FranMorilloAWS 
Request you to review.

File: java/S3ParquetSink/README.md [hunk 1]
```diff
@@ -60,27 +99,102 @@ Flink will [natively and efficiently serialize and deserialize](https://nightlie
...
+```
+ mvn clean compile exec:java  -Dexec.classpathScope="compile" \
+ -Dexec.mainClass="com.amazonaws.services.msf.StreamingJob" 
+
+```
+
+## Deploying using CloudFormation to Amazon Managed Service for Apache Flink
+### Pre-requisite
+1. Kinesis stream and S3 bucket. 
+2. Create subnets and security groups for the Flink application. If you are using private subnets , ensure that VPC endpoint for Kinesis and S3 is created.
```
Comment (nicusX) [inline java/S3ParquetSink/README.md:150]: Same as per #29, VPC networking is not required

File: java/S3ParquetSink/README.md [hunk 2]
```diff
@@ -60,27 +99,102 @@ Flink will [natively and efficiently serialize and deserialize](https://nightlie
...
+3. AWS user credential using which you can create CloudFormation stack from console or CLI.
+
+### Build and deployment
+1. The steps below create stack using `./cloudformation/msf-deploy.yaml`.
+2. The script `deploy.sh` creates the stack using AWS CLI. Ensure that AWS CLI is configured and your user has permissions to create CloudFormation stack.
+3. Alternatively you can deploy from console using `./cloudformation/msf-deploy.yaml` and pass required parameters.
+4. Edit `deploy.sh` to modify  "Region and Network configuration" . Modify following configurations -
+* region= Deployment region
+* SecurityGroup= MSK Security Group.
```
Comment (nicusX) [inline java/S3ParquetSink/README.md:159]: No MSK in this example. Please remove

File: java/S3ParquetSink/cloudformation/msf-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  ApplicationName:
+    Type: String
+  FlinkRuntimeEnvironment:
+    Type: String
+    Default: FLINK-1_15
+  CodeBucketArn:
+    Type: String
+  CodeKey:
+    Type: String
+  SecurityGroup:
```
Comment (nicusX) [inline java/S3ParquetSink/cloudformation/msf-deploy.yaml:12]: No VPC networking required

File: java/S3ParquetSink/cloudformation/msf-deploy.yaml [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+      RuntimeEnvironment: !Ref FlinkRuntimeEnvironment
+      ServiceExecutionRole: !GetAtt FlinkApplicationServiceExecutionRole.Arn
+      ApplicationConfiguration:
+        ApplicationCodeConfiguration:
+          CodeContent:
+            S3ContentLocation:
+              BucketARN: !Ref CodeBucketArn
+              FileKey: !Ref CodeKey
+          CodeContentType: 'ZIPFILE'
+        VpcConfigurations:
```
Comment (nicusX) [inline java/S3ParquetSink/cloudformation/msf-deploy.yaml:48]: As above, no VPC networking

File: java/S3ParquetSink/cloudformation/msf-deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - logs:DescribeLogGroups
+                  - logs:DescribeLogStreams
+                  - logs:PutLogEvents
+                Resource:  "*"
+        - PolicyName: VPCReadOnlyPermissions
```
Comment (nicusX) [inline java/S3ParquetSink/cloudformation/msf-deploy.yaml:151]: As above, no VPC networking

File: java/S3ParquetSink/cloudformation/msf-deploy.yaml [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - ec2:DescribeVpcs
+                  - ec2:DescribeSubnets
+                  - ec2:DescribeSecurityGroups
+                  - ec2:DescribeDhcpOptions
+                Resource: "*"
+        - PolicyName: ENIReadWritePermissions
```
Comment (nicusX) [inline java/S3ParquetSink/cloudformation/msf-deploy.yaml:162]: No VPC networking

File: java/S3ParquetSink/deploy.sh [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+aws s3 ls s3://${bucket}/flink/${jar_name}
+if [ $? -ne 0 ]
+then
+    echo "s3://${bucket}/flink/${jar_name} does not exist"
+    echo "Please execute: build.sh <bucket_name>"
+    echo "Then try again"
+    exit 1
+fi
+
+## Region and Network configuration
```
Comment (nicusX) [inline java/S3ParquetSink/deploy.sh:27]: No VPC networking

**Review by nicusX — CHANGES_REQUESTED** [review]
As commented on #29, please remove VPC networking that is not required in this example

File: java/S3ParquetSink/README.md (hunk 1 above)
Comment (akshayar) [reply java/S3ParquetSink/README.md]: Done. Made CloudFormation changes as well and tested.

File: java/S3ParquetSink/README.md (hunk 2 above)
Comment (akshayar) [reply java/S3ParquetSink/README.md]: Edited this out. Network config not needed anymore.

File: java/S3ParquetSink/cloudformation/msf-deploy.yaml (hunk 3 above)
Comment (akshayar) [reply java/S3ParquetSink/cloudformation/msf-deploy.yaml]: Done and tested.

File: java/S3ParquetSink/cloudformation/msf-deploy.yaml (hunk 4 above)
Comment (akshayar) [reply java/S3ParquetSink/cloudformation/msf-deploy.yaml]: Done and tested.

File: java/S3ParquetSink/cloudformation/msf-deploy.yaml (hunk 6 above)
Comment (akshayar) [reply java/S3ParquetSink/cloudformation/msf-deploy.yaml]: Done and tested.

File: java/S3ParquetSink/deploy.sh (hunk 7 above)
Comment (akshayar) [reply java/S3ParquetSink/deploy.sh]: Done and tested.

**Review by nicusX — APPROVED** [review]
LGTM

**Pull request merged**